modules/trusted-firmware-m/CMakeLists.txt

```diff
@@ -459,6 +473,7 @@ if (CONFIG_BUILD_WITH_TFM)
 
       COMMAND ${PYTHON_EXECUTABLE} ${ZEPHYR_BASE}/scripts/mergehex.py
         -o ${MERGED_FILE}
+        $<TARGET_PROPERTY:tfm,BL1_HEX_FILE>
```

Comment:
This part is a bit confusing, the code flow is like this:

if (NOT BL2)
else if (MCUBOOT_IMAGE_NUMBER == 1 # non-secure and secure is updated as a single merged image)
else # non-secure and secure is updated as separate images

You have only updated the else part, meaning that BL1 support for single image DFU is not updated.
The else and else-if will not be handled correctly in the BL1=ON, BL2=OFF configuration. From what I can tell in the TF-M documentation this is not disallowed. 

So I think it should be:
`if (NOT (CONFIG_TFM_BL2 OR CONFIG_TFM_BL1))`

`$<TARGET_PROPERTY:tfm,BL1_HEX_FILE>` should be added to the `elseif` path.

I'm not sure how `$<TARGET_PROPERTY:tfm,BL1_HEX_FILE>`  behaves when BL1 is not enabled.

I would feel better if they were both (BL1 and BL2):
`$<BOOL:${CONFIG_TFM_BL1},$<TARGET_PROPERTY:tfm,BL1_HEX_FILE>>` for both BL1 and BL2.

Reply:
$<TARGET_PROPERTY:tfm,BL1_HEX_FILE>, if not set is empty